### PR TITLE
Add anchor goto confirm dialog in article link

### PR DESCRIPTION
**Pull request by hueitan**
fixed https://phabricator.wikimedia.org/T241364

styling pending

Annotation 2019-12-31 214625

### CODE REVIEW

File: src/components/Article.js [hunk 1]
```diff
@@ -35,6 +37,10 @@ const ArticleSection = ({
     },
     reference: ({ referenceId }) => {
       showReferencePreview({ reference: references[referenceId], lang })
+    },
+    goto: ({ anchor }) => {
```
Comment (stephanebisson) [inline src/components/Article.js]: I don't think `goto` is appropriate. The other keys represent what the link is. Maybe `section` would be more clear.

File: src/components/Article.js [hunk 2]
```diff
@@ -35,6 +37,10 @@ const ArticleSection = ({
...
+      // @todo styling to be confirmed with design
+      confirmDialog({ message: `Go to Section "${anchor}"?`, onSubmit: () => goToSubpage({ title: anchor }) })
```
Comment (stephanebisson) [inline src/components/Article.js]: This message needs to be localized.

File: src/utils/confirm.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { usePopup } from 'hooks'
+import { ConfirmDialog } from 'components'
+
+export default function ({ message, onSubmit }) {
```
Comment (stephanebisson) [inline src/utils/confirm.js]: Is there a strong reason to have a different style here? All the components are named export of a lambda and this is a default export of a function.

File: src/components/ConfirmDialog.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  useSoftkey('ConfirmDialog', {
+    right: i18n.i18n('softkey-close'),
+    onKeyRight: close,
+    center: i18n.i18n('softkey-ok'),
+    onKeyCenter: () => { onSubmit(); close() }
+  }, [])
+
+  return (
+    <div class='confirm-dialog'>
+      <div class='item'>
```
Comment (stephanebisson) [inline src/components/ConfirmDialog.js]: Why do we have this hierarchy or `.item > .info > .title` to display one string?

File: src/utils/confirm.js (hunk 3 above)
Comment (hueitan) [reply src/utils/confirm.js]: true, this just returns a single function.

File: src/hooks/useArticleLinksNavigation.js [hunk 5]
```diff
@@ -106,15 +107,18 @@ const makeLinkClickEvent = link => {
...
+    const referenceId = link.getAttribute('href').slice(1)
     return { type: 'reference', referenceId }
   }
 
   if (link.getAttribute('rel') === 'mw:ExtLink') {
     return { type: 'external', href: link.href }
   }
+
+  if (link.hash) {
+    return { type: 'section', anchor: link.textContent }
```
[the remark this answers is not in the document]
Comment (hueitan) [reply src/hooks/useArticleLinksNavigation.js]: good catch! text and anchor might be different!

File: src/components/ConfirmDialog.js (hunk 4 above)
Comment (hueitan) [reply src/components/ConfirmDialog.js]: good catch, i was copying this from listview, and we don't need item as the class name!

I also make the change to fit longer section name
Capture

File: i18n/qqq.json [hunk 6]
```diff
@@ -10,6 +10,7 @@
 	"app-title": "Title of the application. Shown in the app store and the app launcher.",
 	"app-subtitle": "Subtitle of the application. Shown in the app store.",
 	"app-description": "Description of the application. Shown in the app store.",
+	"confirm-section": "Confirm message switch to the selected section page. Shown in the Article page.",
```
[the remark this answers is not in the document]
Comment (hueitan) [reply i18n/qqq.json]: done.

**Pull request merged**